src/validation.cpp

```diff
@@ -3544,13 +3558,41 @@ uint64_t CalculateCurrentUsage()
 }
 
 /* Prune a block file (modify associated database entries)*/
-void PruneOneBlockFile(const int fileNumber)
+void PruneOneBlockFile(const int fileNumber, CValidationState &state)
 {
+    AssertLockHeld(cs_main);
     LOCK(cs_LastBlockFile);
 
+    int64_t txindex_migration_totaltime = 0;
     for (const auto& entry : mapBlockIndex) {
         CBlockIndex* pindex = entry.second;
         if (pindex->nFile == fileNumber) {
+            if (fTxIndex && (pindex->nStatus & BLOCK_HAVE_DATA)) {
+                int64_t time_start = GetTimeMicros();
+                // migrate pruned txindex data
+                // txindex entries of pruned blocks will move its CDiskTxPos value to:
+                //        nFile = INT_MAX <-- indicator that this block is no longer available
+                //        nPos = block-height in chainActive (assumed reorg safe since prune min target is 288 blocks)
+                //        nTxOffset = index of transaction position in block (not to confused with data offset)
+                LogPrintf("txindex block pruning: %s migrate txindex data from block %s\n", __func__, pindex->GetBlockHash().GetHex());
+                CBlock block;
+                if (!ReadBlockFromDisk(block, pindex, Params().GetConsensus())) {
+                    AbortNode(state, "Failed to read block");
+                    return;
+                }
+                std::vector<std::pair<uint256, CDiskTxPos>> vPos;
+                unsigned int txpos = 0;
+                for(const auto& tx : block.vtx)
```

Comment:
Missing space after `for`. `{` in this line.

Reply:
clang-format will fix this and maybe more.